src/parallel/parallel.cpp

```diff
@@ -193,10 +359,39 @@ bool
 ParallelRobinHoodHashTable::locked_insert(ParallelBucket &entry_to_insert, size_t swap_index)
 {
     // TODO
+    //size_t swap_index = entry_to_insert.offset; //? i thnk its redoing it bc of possible contention but its locked so??
+    
+    ThreadManager manager = this->get_thread_lock_manager(); 
+    while(true) {
+        ParallelBucket &swapped_entry = this->do_atomic_swap(entry_to_insert, swap_index);
```

Comment:
```suggestion
        entry_to_swap = this->do_atomic_swap(entry_to_swap, swap_index);
```
Original algorithm mutates `swapped_entry` in each iteration of the loop.